jwst/resample/resample_spec.py

```diff
@@ -65,11 +65,10 @@ def __init__(self, input_models, output=None, **pars):
         self.blank_output = None
 
         # Define output WCS based on all inputs, including a reference WCS
-        # wcslist = [m.meta.wcs for m in self.input_models]
         self.output_wcs = self.build_interpolated_output_wcs()
-        self.blank_output = datamodels.DrizProductModel(self.data_size)
-
-        self.blank_output.update(datamodels.ImageModel(self.input_models[0]._instance))
+        self.blank_output = datamodels.SlitModel(self.data_size)
+        self.input_models[0].meta.bunit_err = None
```

Comment:
This line is causing the BUNIT keyword to disappear from the ERR extension header in the output cal file when processing something like NIRSpec FS or MOS (in MultiSlitModel) through `calwebb_spec2`, because the cal file hasn't been saved yet from the input_model (it gets saved at the very end of the pipeline).

If I remove this line and instead add the following 2 lines after line 71:
```
self.blank_output.err = None
self.blank_output.meta.bunit_err = None
```
that retains the BUNIT keyword for the ERR extension in the output cal file, but unfortunately is still causing an ERR extension to show up in the resampled s2d file (along with SCI, WHT, and CON) and in this case it even has non-zero size. So why is that still showing up when the err array is set to None?

@stscieisenhamer @jdavies-st

Reply:
Instead of setting them to None:

```
del self.blank_output.meta.bunit_err
del self.blank_output.err
```

Reply:
Thanks James. 
del self.blank_output.meta.bunit_err
also deletes the err array.

Reply:
Ding! Ding! Ding! We have a winner. Simply deleting bunit_err seems to be enough. When I try to delete the err array itself, I get a "Attribute 'err' missing" error, but maybe that's because I deleted the default values for the all the data arrays in the slitdata.schema, so it doesn't get created to begin with. And now it no longer shows up in the output s2d file.

Reply:
Yeah, the zero-size error array only get created when written to disk.  It needs to be created because on writing, the `bunit_err` in the ASDF tree gets mapped to that array.  If you don't have `bunit_err` in the tree, the extension doesn't get written.  So as long as the model hasn't been written out, deleting `bunit_err` is enough.